### ISSUE TITLE
Uncertainty Propagation Addendum

### ISSUE THREAD

**Issue by SmithUoG** [dmzuckerman/Sampling-Uncertainty#49]
I'm late to this discussion, but I'm a strong supporter of the topic and the project, and I have a couple of comments on Section 7.4.

I think that the section is a little "thin".   Although the density example of  Eqs. (20) and (21) is fine, it's a bit trivial, and more examples could be given.  Here are 2.

1. The example of the propagation of the uncertainty in enthalpy to that in Cp is given in our paper in JCP, 147, 034508(2017).  

Over a given T range at fixed P, one can fit h(T) = a+b*T+c*T^2 and then differentiate to get Cp.  The uncertainty in Cp arises from the uncertainties in the regression coefficients b and c.  The linear approximation gives 
s^2(Cp) = s^2(b) + 4*T^2*s^2(c) +4*T*cov(b,c)
The covariance matrix is available from the output of a typical linear regression algorithm (e.g., in Excel).

2.  The example of the uncertainty in the solubility in an aqueous solution at a given (T,P), which is calculated by equating the solute solid chemical potential mu_s to its chemical potential in solution, mu(m), where m is the molality.  This is described in Eqs,. (21)-(24) in our review paper in Molec. Phys., 114, 1665-1690 (2016).  

This is an example of uncertainty propagation via the solution of a nonlinear equation.  In this particular case, the goal is to estimate the uncertainty in the solution m_s of the nonlinear equation

mu(m) = mu_s

Both mu_s and mu(m) have simulation uncertainties in this case.

**Comment by agrossfield** [comment]
This is an interesting suggestion, but I think an additional example would make this section too long relative to its importance.  Moreover, I think the second example, while really cool, is too advanced for the goals of this manuscript, in the sense that it would force us to address a simulation technique that our intended audience (people relatively new to molecular simulation) likely are not familiar with.  

What do the other authors think?

**Comment by dwsideriusNIST** [comment]
@agrossfield I completely agree that it's a beneficial idea. But Paul and I decided to ignore it for the time being so that the paper v1.0 is finished in a timely manner.

We can do this (perhaps invite @SmithUoG to be an author) in a future revision or as an accompanying Jupyter notebook.

**Comment by SmithUoG** [comment]
OK with me too, since I'm a "Johnny Come Lately" to this party anyway.

But let me comment that error/uncertainty propagation is a HUGELY important and often overlooked topic, not only in molecular simulation, but in all areas of science where data analysis is involved.  So I would claim to the contrary that "the section is too short relative to its importance" :-)

Also, the examples are "generic" in the sense that the basic principles are independent of any simulation methodology; they are simply general examples of uncertainty propagation.  Also, the first involves a pretty simple simulation procedure.  If we applied it to simulations of U over a T range to get Cv by fitting U(T)=a+b*T, then it would be simpler still; the uncertainty of Cv would involve only the uncertainty of the regression coefficient b.  Using a quadratic expression nicely brings in the covariance of the parameters b and c.

On another related matter, I think it would be good to refer to the following 2 papers for what I think is a good discussion of the principles of Uncertainty Analysis in chemistry.

@article{Nicholls2014,
   author = {Nicholls, A.},
   title = {Confidence limits, error bars and method comparison in molecular modeling. {P}art 1: {T}he calculation of confidence intervals},
   journal = {J Comput Aided Mol Des},
   volume = {28},
   number = {9},
   pages = {887-918},
   year = {2014},
}

**Comment by SmithUoG** [comment]
Oops, I forgot the second paper.  Here it is.

@article{Nicholls2016,
   author = {Nicholls, A.},
   title = {Confidence limits, error bars and method comparison in molecular modeling. Part 2: comparing methods},
   journal = {J Comput Aided Mol Des},
   volume = {30},
   number = {2},
   pages = {103-126},
   year = {2016},
}

**Comment by dmzuckerman** [comment]
@SmithUoG thanks for the references and the suggestions.  I will try to look over all this in the next couple of days.  I agree with you that understanding uncertainty in all derived quantities is very important.  This is one reason why bootstrapping approaches are valuable ... they implicitly capture the uncertainty propagation - @mangiapasta do you agree?

**Comment by davidlmobley** [comment]
Note also the idea of having supporting documents (or Jupyter notebooks) in the repo itself which would be linked to from the main text is probably a rather good one and a way to introduce more concepts in the paper without having to expand it a lot. :)
